### PR TITLE
Add rosdistro version to upstream ws cache prefix

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Hopefully fix: https://github.com/ros-planning/moveit2/issues/783 and https://github.com/ros-planning/moveit2/issues/947

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -66,14 +70,31 @@ jobs:
...
+              echo "$(curl -sL "https://api.github.com/repos/ros/rosdistro/tags?per_page=100" \
+              | grep -o "$1/[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}" \
+              | cut -d'/' -f2 \
+              | uniq \
+              | sort -t- -k1n -k2n -k3n -z | head -n 1)"
+          }
+          ROS_DISTRO_VERSION=`latest-release-date $ROS_DISTRO`
+          echo "The latest version for $ROS_DISTRO is $ROS_DISTRO_VERSION"
+          echo set-output name=ROS_DISTRO_VERSION::${ROS_DISTRO_VERSION}
+          echo ::set-output name=ROS_DISTRO_VERSION::${ROS_DISTRO_VERSION}
```
Comment (tylerjw) [inline .github/workflows/ci.yaml:89]: What does the :: prefix do and why do we need both of these lines?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -66,14 +70,31 @@ jobs:
...
+        env:
+          ROS_DISTRO: ${{ matrix.env.ROS_DISTRO }}
+        run: |
+          latest-release-date() {
+              echo "$(curl -sL "https://api.github.com/repos/ros/rosdistro/tags?per_page=100" \
+              | grep -o "$1/[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}" \
+              | cut -d'/' -f2 \
+              | uniq \
+              | sort -t- -k1n -k2n -k3n -z | head -n 1)"
+          }
```
Comment (tylerjw) [inline .github/workflows/ci.yaml:85]: Should put this in another repo as @rhaschke did with the lcov one below?  Then we could easily reuse this in all our CI jobs.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (JafarAbdi) [reply .github/workflows/ci.yaml]: The first one is just a debugging message, the second one for setting a step output value

**Review by tylerjw — APPROVED** [review]
I like this much better.  Thank you for moving the code into your own action.

**Review by vatanaksoytezer — APPROVED** [review]
That looks good to me, one quick question: how do we verify this resolves our problem? If we cannot it seems moveit2_tutorials Foxy CI seems broken the same way and we can use it to verify it fixes our issue.

**Pull request merged**